src/BackupDestination/Backup.php

```diff
@@ -27,24 +36,38 @@ public function path(): string
 
     public function exists(): bool
     {
-        return $this->disk->exists($this->path);
+        if ($this->exists === null) {
+            $this->exists = $this->disk->exists($this->path);
+        }
+
+        return $this->exists;
     }
 
     public function date(): Carbon
     {
-        return Carbon::createFromTimestamp($this->disk->lastModified($this->path));
+        if ($this->date === null) {
+            $this->date = Carbon::createFromTimestamp($this->disk->lastModified($this->path));
+            $this->exists = true;
```

Comment:
Do we really need to set `exists` here?

Reply:
I think so yes. This will prevent extra calls. E.g. the IO operations when running `it_can_remove_old_backups_until_using_less_than_maximum_storage` are:
```date   mysite/test1.zip
date   mysite/test2.zip
date   mysite/test3.zip
date   mysite/test4.zip
date   mysite/test2.zip
date   mysite/test3.zip
date   mysite/test4.zip
date   mysite/test2.zip
date   mysite/test3.zip
date   mysite/test4.zip
date   mysite/test2.zip
date   mysite/test3.zip
date   mysite/test4.zip
date   mysite/test2.zip
date   mysite/test3.zip
date   mysite/test4.zip
exists mysite/test2.zip
date   mysite/test2.zip
exists mysite/test3.zip
date   mysite/test3.zip
exists mysite/test4.zip
date   mysite/test4.zip
exists mysite/test2.zip
exists mysite/test3.zip
exists mysite/test4.zip
size   mysite/test2.zip
exists mysite/test2.zip
size   mysite/test3.zip
exists mysite/test3.zip
size   mysite/test4.zip
exists mysite/test4.zip
size   mysite/test1.zip
exists mysite/test1.zip
exists mysite/test2.zip
exists mysite/test3.zip
exists mysite/test4.zip
exists mysite/test2.zip
exists mysite/test3.zip
size   mysite/test2.zip
exists mysite/test2.zip
size   mysite/test3.zip
exists mysite/test3.zip
size   mysite/test1.zip
exists mysite/test1.zip
exists mysite/test2.zip
exists mysite/test3.zip
exists mysite/test2.zip
size   mysite/test2.zip
exists mysite/test2.zip
size   mysite/test1.zip
exists mysite/test1.zip
date   mysite/test1.zip
date   mysite/test2.zip
exists mysite/test1.zip
exists mysite/test2.zip
size   mysite/test1.zip
exists mysite/test1.zip
size   mysite/test1.zip
exists mysite/test1.zip
size   mysite/test2.zip
exists mysite/test2.zip
date   mysite/test1.zip
date   mysite/test2.zip
date   mysite/test1.zip
date   mysite/test2.zip
size   mysite/test1.zip
exists mysite/test1.zip
size   mysite/test2.zip
exists mysite/test2.zip
```

as you can see, `exists()` calls are made after `date()` calls

Reply:
I think the code is simpler if each method just caches its own result. I don't really care if we have one call extra that we could prevent. Could would be much easier to follow and performance will be better than it currently is.

Could you make this change?

Reply:
sure, done